### ISSUE TITLE
Clarify which inputs to an op are input values vs attributes in the spec

### ISSUE THREAD

**Issue by bchetioui** [openxla/stablehlo#1683]
I was reading through the spec, and I noticed that there is no explicit mention of what is an op attribute vs what is an op input value in the specification of each op. E.g., for the [slice op](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#slice), there are four listed inputs, but only one is an op input value (vs an input attribute). All the others must be known at compile-time, but there is no way to know this from looking at the op: all of them are listed as tensors.

I think the spec would be enhanced if we either

a. added an "Attributes" section for each op under "Inputs", split from the "Input values" (like the distinction that is made in the [general Operations discussion](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#slice)
b. added the word "attribute" in the description of each argument that is meant to be known statically.

**Comment by GleasonK** [comment]
This makes sense, triaging to spec workstream. Another option I could imagine it to change the labeling, `(I1), (I2)` to reflect operand vs attribute `(A2)` for attr for example.

**Comment by burmako** [comment]
@bchetioui Yes, in the beginning, we were also planning to split Operands and Attributes, but then decided to merge them into just Inputs and express the distinction in types. E.g. if we take the slice example, `operand` is "tensor or per-tensor quantized tensor", but `start_indices` is "1-dimensional tensor **constant** of type si64". Would this convention work on your side?

**Comment by bchetioui** [comment]
@burmako I did notice that distinction eventually, but as noted in my earlier edit to the initial comment, I don't find "constant" to be very explicit about the fact that this is known at compile-time (although if you think about it deeper, it is true). Obviously, it's ultimately your call, but I'd find it **much** more intuitively understandable if the word attribute at least appeared in the description (e.g. instead of constant).

Perhaps you could also add information that "attributes are known at compile-time and are designated as constant tensors in the following" in the general discussion on Operations?
